### PR TITLE
server: add logging for action handlers

### PR DESCRIPTION
**Pull request by ecthiender**
### Description

Added logs for action handlers. Right now it logs request and response sizes to and from the action handler.

A sample log line:
```json
{                                                                                                                                                      
  "type": "action-handler-log",                                                                                                                            
  "timestamp": "2020-07-28T22:10:35.982+0530",                                                                                                         
  "level": "info",                                                                                                                                     
  "detail": {                                                                                                                                          
    "response_size": 36,                                                                                                                               
    "request_size": 102                                                                                                                                
  }                                                                                                                                                    
}
```

### Changelog

- [x] `CHANGELOG.md` is updated with user-facing content relevant to this PR. If no changelog is required, then add the `no-changelog-required` label.

### Affected components

- [x] Server

### Related Issues

https://github.com/hasura/graphql-engine-pro/issues/431

### Solution and Design

Add logging inside the `callWebhook` function in `Hasura.GraphQL.Resolve.Action` module. This currently logs the request and repsonse sizes to and from the action webhook. 

The simple change is in this module (`Hasura.GraphQL.Resolve.Action`), rest of the changes are required to pass the logger to appropriate functions.

#### Catalog upgrade

Does this PR change Hasura Catalog version?
- [x] No
- [ ] Yes
  - [ ] Updated docs with SQL for downgrading the catalog 

#### Metadata

Does this PR add a new Metadata feature?
- [x] No
- [ ] Yes
  - Does `run_sql` auto manages the new metadata through schema diffing?
    - [ ] Yes
    - [ ] Not required
  - Does `run_sql` auto manages the definitions of metadata on renaming?
    - [ ] Yes
    - [ ] Not required
  - Does `export_metadata`/`replace_metadata` supports the new metadata added?
    - [ ] Yes
    - [ ] Not required

#### GraphQL
- [x] No new GraphQL schema is generated
- [ ] New GraphQL schema is being generated:
   - [ ] New types and typenames are correlated
   
   

#### Breaking changes

- [x] No Breaking changes
- [ ] There are breaking changes:

  1. Metadata API

     Existing `query` types:
     - [ ] Modify `args` payload which is not backward compatible
     - [ ] Behavioural change of the API
     - [ ] Change in response `JSON` schema
     - [ ] Change in error code
     

  2. GraphQL API

     Schema Generation:
     
     
     - [ ] Change in any `NamedType`
     - [ ] Change in table field names
     

     Schema Resolve:-
     
     - [ ] Change in treatment of `null` value for any input fields 
     

  3. Logging

     - [ ] Log `JSON` schema has changed
     - [ ] Log `type` names have changed

### CODE REVIEW

**Comment by tirumaraiselvan** [comment]
@ecthiender Does this close this issue? https://github.com/hasura/graphql-engine/issues/4171

**Comment by ecthiender** [comment]
> @ecthiender Does this close this issue? #4171

@tirumaraiselvan somewhat. It doesn't do the comprehensive logging outlined by #4171. Currently it only logs the request/response size. (As shown in the PR description). We can modify this to support all of the requirements of #4171 but that would increase the scope of our original issue wrt cloud-sprint. I'll let @paf31 and you decide. And if the extra work is required on this PR, I can do that.

**Comment by tirumaraiselvan** [comment]
I think it's not required right now. #4171 can be treated separately as a `debug` log line.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
 (Add entries here in the order of: server, console, cli, docs, others)
 
 - console: update sidebar icons for different action and trigger types
+- server: add logs for action handlers
```
Comment (tirumaraiselvan) [inline CHANGELOG.md:10]: When you fix the conflict in this file, could you also move the server entry to top.

File: server/src-lib/Hasura/Logging.hs [hunk 2]
```diff
@@ -92,6 +94,7 @@ instance J.FromJSON (EngineLogType Hasura) where
     "websocket-log" -> return ELTWebsocketLog
     "query-log" -> return ELTQueryLog
     "livequery-poller-log" -> return ELTLivequeryPollerLog
+    "action-handler" -> return ELTActionHandler
```
Comment (tirumaraiselvan) [inline server/src-lib/Hasura/Logging.hs:97]: for consistency, should we call this `action-handler-log`?

**Comment by ecthiender** [comment]
If you're saying we can have two different log-types (when #4171 is merged), then I agree that's the most low-friction approach. But thinking about it a bit more, isn't it redundant that there would be two log-types which log seemingly similar information? But maybe not? They have two different functions?

Ideally, I think this behaviour makes sense - when log-level is `info` it logs request/response size (to/from the call to action handler), and when `log-level` is `debug` it does more comprehensive logging. But our current logging infra doesn't help with this. Currently one log-type is fixed to a specific log-level.

**Comment by tirumaraiselvan** [comment]
> If you're saying we can have two different log-types (when #4171 is merged), then I agree that's the most low-friction approach. But thinking about it a bit more, isn't it redundant that there would be two log-types which log seemingly similar information? But maybe not? They have two different functions?
> 
> Ideally, I think this behaviour makes sense - when log-level is `info` it logs request/response size (to/from the call to action handler), and when `log-level` is `debug` it does more comprehensive logging. But our current logging infra doesn't help with this. Currently one log-type is fixed to a specific log-level.

Actually, I think the ideal behaviour would be to print `info` log line (as given in this PR) and another `debug` log line as given in #4171. So that log lines are always "additive".

**Pull request merged**